suse_migration_services/prechecks/kernels.py

```diff
@@ -54,24 +69,63 @@ def multiversion_and_multiple_kernels():
                         'Please ensure it is set as:\n'
                         "'multiversion.kernels = latest,running'")
         elif set(kernel_multi_version_settings.split(',')) != {'running', 'latest'}:
-            log.warning('The config option multiversion.kernels is not set '
-                        'correctly in /etc/zypp/zypp.conf. It is currrently '
-                        'set as \n%s.\nPlease ensure it is set as:\n'
-                        "'multiversion.kernels = latest,running'",
-                        kernel_multi_version_settings)
+            if fix:
+                log.info("The '--fix' option was provided, "
+                         "setting 'multiversion.kernels = latest,running' ")
+                existing = 'multiversion.kernels = ' + kernel_multi_version_settings
+                correction = 'multiversion.kernels = latest,running'
+                sed_arg = 's/' + existing + '/' + correction + '/'
+                try:
+                    Command.run(["sed", "-i", sed_arg, Defaults.get_zypp_config_path()])
+                except DistMigrationCommandException:
+                    log.error("ERROR: Unable to update /etc/zypp/zypp.conf")
+            else:
+                log.warning("The config option multiversion.kernels is not set "
+                            "correctly in /etc/zypp/zypp.conf. It is currrently "
+                            "set as:\n'multiversion.kernels = %s'.\n"
+                            "Please ensure it is set as:\n"
+                            "'multiversion.kernels = latest,running'",
+                            kernel_multi_version_settings)
+
+
+def multiple_kernels_installed(fix, target, log):
+    """Check for multiple kernels installed"""
 
     kernel_type = 'kernel-default'
-    running_kernel = Command.run(["uname", "-r"])
+    running_kernel_path = '/' + Defaults.get_target_kernel()
+    rpm_command = ["rpm", "-qa", kernel_type]
+    if target:
+        running_kernel_path = Defaults.get_system_root_path() + running_kernel_path
+        rpm_command = \
+            [
+                "chroot",
+                Defaults.get_system_root_path(),
+                "rpm",
+                "-qa",
+                kernel_type
+            ]
+
+    running_kernel = os.readlink(running_kernel_path)
 
-    if 'azure' in running_kernel.output:
+    if 'azure' in running_kernel:
         kernel_type = 'kernel-azure'
 
-    installed_kernels = Command.run(["rpm", "-qa", kernel_type])
-    multiple_kernels = len(installed_kernels.output.splitlines()) > 1
+    installed_kernels = Command.run(rpm_command).output.splitlines()
```

Comment:
why did we change the use of `multiple_kernels` to go back to len(....) > 1 in the if ? 
did we not change it before ?

Reply:
Jesus,  I need to be able to print the list lof installed_kernels for the user. if I do 

```
    installed_kernels = Command.run(rpm_command).output.splitlines()
    multiple_kernels = len(installed_kernels) > 1
```    
that seems like declaring an extra variable.  Perhaps I did not understand fully, if so can you please explain.